build-support/bin/check_header.py

```diff
@@ -20,8 +20,14 @@
 
 EXPECTED_NUM_LINES = 3
 
-_current_year = str(datetime.datetime.now().year)
-_current_century_regex = re.compile(r'20\d\d')
+CURRENT_YEAR = str(datetime.datetime.now().year)
+CURRENT_CENTURY_REGEX = re.compile(r'20\d\d')
+
+PY2_DIRECTORIES = {
```

Comment:
This all looks good but if I haven't pointed it out already, we already have a rule that does all this introduced in #7515. It already has enough flexibility to apply header checks to subsets of paths although it uses whitelists and could stand support for blacklists to make path specification more concise and less brittle for cases like these. We use the rule in the toolchain codebase; so I've filed #7942 to add blacklists and circling back to replace this hand-rolled check with use of the rule.

Reply:
Ack. Always a fan of less code.